pages/_app.tsx

```diff
@@ -16,26 +16,47 @@ import "src/sections/community/styles.scss";
 import "src/sections/home/styles.scss";
 import "src/sections/sponsorship/styles.scss";
 
+import { useEffect } from 'react'
+import { useRouter } from 'next/router'
 import NavigationBar from "src/components/NavigationBar";
 import Footer from "src/components/Footer";
+import * as ga from "src/utils/analytics"
 
-const MyApp = ({ Component, pageProps }) => (
-  <>
-    <NavigationBar />
-    <div
-      style={{
-        margin: `0 auto`,
-        maxWidth: 960,
-        minHeight: 960,
-        padding: `85px 0`,
-      }}
-    >
-      <main>
-        <Component {...pageProps} />
-      </main>
-    </div>
-    <Footer />
-  </>
-);
+
+const MyApp = ({ Component, pageProps }) => {
+  const router = useRouter();
+  useEffect(() => {
+    const handleRouteChange = (url) => {
+      ga.pageview(url)
+    }
+    //When the component is mounted, subscribe to router changes
+    //and log those page views
+    router.events.on('routeChangeComplete', handleRouteChange)
+
+    // If the component is unmounted, unsubscribe
+    // from the event with the `off` method
+    return () => {
+      router.events.off('routeChangeComplete', handleRouteChange)
+    }
+  }, [router.events])
+  return (
+    <>
+      <NavigationBar />
+      <div
+        style={{
+          margin: `0 auto`,
+          maxWidth: 960,
+          minHeight: 960,
+          padding: `85px 0`,
+        }}
+      >
+        <main>
+          <Component {...pageProps} />
```

Comment:
idea: im wondering if it would make more sense to move this to a Layout component that wraps all of the pages individually so we have a little more flexibility

Reply:
I think `_app.tsx` is meant to replace the idea of a layout component completely, but I think if we end up having more different layouts, that might be worth doign

Reply:
If there's only one layout, no point in making separate containers unless necessary. I feel like this is good for now.